### ISSUE TITLE
panic: runtime error: slice bounds out of range [3:1]

### ISSUE THREAD

**Issue by jasonblewis** [jmhobbs/wp-to-11ty#2]
Hi,

I tested with  wp-to-11ty-v0.0.2-linux-amd64.tar.gz

When I try and run wp-to-11ty on my export.xml, I get the following error:

```
11:35:26 Writing scaffolding files...
11:35:26 Writing out pages and posts...
11:35:26 Skipping draft: AAA
11:35:26 Skipping draft: BBB CCC
11:35:26 Skipping draft: dddd eeee.
11:35:26 Skipping draft: test google maps
panic: runtime error: slice bounds out of range [3:1]

goroutine 1 [running]:
main.pathToFileSystem({0xc000023397?, 0x21?})
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.3/main.go:230 +0x1b0
main.writeOutPage({_, _}, {{0xc0001ff6c0, 0x8}, {0xc000023380, 0x21}, {0xc00048fda0, 0x1f}, {0xc0001ff708, 0x5}, ...})
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.3/main.go:148 +0x8d
main.writePagesAndPosts({0xc00001a8a0, 0x4}, 0xe?)
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.3/main.go:92 +0x1f8
main.main()
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.3/main.go:69 +0x40b
```

I suspect it could be due to some funny headings in my xml. like for instance (from the xml): 

`<title><![CDATA[\(\LaTeX\) Equations in Wordpress using MathJax]]></title>`

**Comment by jmhobbs** [comment]
Interesting!  It actually looks from the stack trace that it's failing on a `<link>` tag.

Do you have any `<link>` with unusual (or very short) URL Paths?  Perhaps a page set as the blog home page?

**Comment by jmhobbs** [comment]
Ok, so I reproduced this by setting one of the `<link>` fields on an `<item>` to my domain with no path.

I patched that and will merge it in a moment, then if you can give it a shot that would be great.

**Comment by jmhobbs** [comment]
It looks like you installed with `go install`, so you should be able to grab the update with `go install github.com/jmhobbs/wp-to-11ty@v0.0.4`

**Comment by jasonblewis** [comment]
Thanks, I tested with v0.0.4 but I get a similar error: 

```
$ wp-to-11ty export.xml 
14:34:21 Writing scaffolding files...
14:34:21 Writing out pages and posts...
panic: runtime error: slice bounds out of range [4:2]

goroutine 1 [running]:
main.pathToFileSystem({0xc00001c427?, 0x47?})
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.4/main.go:230 +0x1b0
main.writeOutPage({_, _}, {{0xc0000223f0, 0x2e}, {0xc00001c410, 0x47}, {0xc0000204e0, 0x1f}, {0xc00001ad68, 0x5}, ...})
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.4/main.go:148 +0x8d
main.writePagesAndPosts({0xc00001a8a0, 0x4}, 0xe?)
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.4/main.go:92 +0x1f8
main.main()
	/home/jason/go/pkg/mod/github.com/jmhobbs/wp-to-11ty@v0.0.4/main.go:69 +0x40b
```

**Comment by jasonblewis** [comment]
is there some way to get wp-to-11ty to output some logs about what it's doing before the crash? or which entry it is parsing when it crashes?

**Comment by jmhobbs** [comment]
Sorry that didn't work. Nothing currently, but I'll put in some verbose logging.  Thanks for your patience!

**Comment by jmhobbs** [comment]
Ah!  I didn't tag the right commit 😅 
![Screenshot 2023-05-10 at 10 06 35 AM](https://github.com/jmhobbs/wp-to-11ty/assets/115059/e102f0f8-6f41-4cee-bd6a-8bd5da22d070)
I'll fix up the v0.0.4 tag, but you could try installing the exact commit, `go install github.com/jmhobbs/wp-to-11ty@3b4377f1e41b1b01089a5a58a2157f16f2f0befd`

**Comment by jasonblewis** [comment]
Thanks, building the latest seems to have fixed that issue, Thank you
